### PR TITLE
Update the Method Call metric usage

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Update the MethodCall metric code to allow multiple usage with different operation names + cleanup some SR code

### CODE REVIEW

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScope.kt [hunk 1]
```diff
@@ -47,11 +47,13 @@ interface FeatureScope {
      * @param callerClass  name of the class calling the performance measurement.
      * @param metric name of the metric that we want to measure.
      * @param samplingRate value between 0-100 for sampling the event.
+     * @param operationName the name of the operation being measured
      * @return a PerformanceMetric object that can later be used to send telemetry, or null if sampled out
      */
     fun startPerformanceMeasure(
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScope.kt:53]: side note: I added some comments to the original PR https://github.com/DataDog/dd-sdk-android/pull/1940#pullrequestreview-2037499101, but they were too late. I think in the future we should refactor this a bit a remove this method from the `FeatureScope`.

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScopeExt.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Measures the execution time for the given block and report it as a MethodCall telemetry metric.
+ * @param R the type of the result of the operation
+ * @param callerClass the class calling the measured method
+ * @param operationName the operationName to report in the metric
+ * @param samplingRate the sampling rate for the metric
+ * @param operation the operation to report
+ */
+fun <R : Any?> FeatureScope?.measureMethodCallPerf(
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScopeExt.kt:19]:
```suggestion
@InternalApi
fun <R : Any?> FeatureScope?.measureMethodCallPerf(
```

File: dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScopeExt.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        val metric = startPerformanceMeasure(
+            callerClass = callerClass.name,
+            metric = TelemetryMetricType.MethodCalled,
+            samplingRate = samplingRate,
+            operationName = operationName
+        )
+
+        val result = operation()
+
+        val isSuccessful = (result != null) && ((result !is Collection<*>) || result.isNotEmpty())
```
Comment (0xnm) [inline dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/feature/FeatureScopeExt.kt:37]: why generally speaking empty collection shouldn't be considered as a successful result? but probably for the methods we are going to cover with this call it makes sense.

File: features/dd-sdk-android-session-replay/build.gradle.kts [hunk 4]
```diff
@@ -52,6 +52,7 @@ android {
 
 dependencies {
     api(project(":dd-sdk-android-core"))
+    api(project(":features:dd-sdk-android-trace"))
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/build.gradle.kts:55]: blocking: seems like unnecessary change? I don't see any `com.datadog.android.trace` or `com.datadog.trace` usages in this PR

File: features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/listener/WindowsOnDrawListener.kt [hunk 5]
```diff
@@ -31,59 +27,44 @@ internal class WindowsOnDrawListener(
...
-        val performanceMetric = sessionReplayFeature?.startPerformanceMeasure(
-            callerClass = this.javaClass.name,
-            metric = TelemetryMetricType.MethodCalled,
-            samplingRate = methodCallTelemetrySamplingRate
-        )
-
-        val nodes = views
-            .mapNotNull {
+        val nodes = sessionReplayFeature.measureMethodCallPerf(
+            javaClass,
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/listener/WindowsOnDrawListener.kt:58]: should this be in the consumer rules to prevent name obfuscation?

File: dd-sdk-android-core/api/apiSurface [hunk 6]
```diff
@@ -43,6 +43,7 @@ interface com.datadog.android.api.InternalLogger
   fun log(Level, Target, () -> String, Throwable? = null, Boolean = false, Map<String, Any?>? = null)
   fun log(Level, List<Target>, () -> String, Throwable? = null, Boolean = false, Map<String, Any?>? = null)
   fun logMetric(() -> String, Map<String, Any?>)
+  fun startPerformanceMeasure(String, com.datadog.android.core.metrics.TelemetryMetricType, Float, String): com.datadog.android.core.metrics.PerformanceMetric?
```
Comment (0xnm) [inline dd-sdk-android-core/api/apiSurface:46]: nice change!

**Review by 0xnm — APPROVED** [review]

File: dd-sdk-android-core/api/apiSurface (hunk 6 above)
Comment (xgouchet) [reply dd-sdk-android-core/api/apiSurface:46]: Yeah It makes more sense, and makes it easier to use too
